devito/ir/support/vector.py

```diff
@@ -0,0 +1,355 @@
+from collections import OrderedDict
+
+from sympy import Basic, true
+
+from devito.tools import as_tuple, is_integer, memoized_meth
+from devito.types import Dimension
+
+__all__ = ['Vector', 'LabeledVector', 'vmin', 'vmax']
+
+
+class Vector(tuple):
+
+    """
+    A representation of an object in Z^n.
```

Comment:
Why `Z`? (I'm assuming it stands for the integers).

Reply:
Correct, integers

It's Z (and not R, for example) because certain operations that this class provides are guaranteed to work (or are well-defined) only over integers 

BTW, note that this class has been in master for ages, I only moved it here from a different file

Reply:
Oki.

> BTW, note that this class has been in master for ages, I only moved it here from a different file

Yeah, I realize that, I'd just not been through it before.